File: calendar/app/src/main/java/com/example/calendar/Event.java
package com.example.calendar;

import java.util.Calendar;

public class Event {
    private String Title;
    private Calendar calendar;

    Event(String t, Calendar cal){
        this.Title = t;
        this.calendar = cal;
    }

    public String getTitle() {
        return Title;
    }

    public void setTitle(String title) {
        Title = title;
    }


    public Calendar getCalendar() {
        return calendar;
    }
}

File: calendar/app/src/main/java/com/example/calendar/Event.kt
package com.example.calendar

import com.alamkanak.weekview.WeekViewDisplayable
import com.alamkanak.weekview.WeekViewEvent
import me.jlurena.revolvingweekview.WeekViewEvent
import java.util.*

data class Event(
    val id: Long,
    val title: String,
    val startTime: Calendar,
    val endTime: Calendar,
    val location: String,
    val color: Int,
    val isAllDay: Boolean,
    val isCanceled: Boolean
) : WeekViewDisplayable<Event> {

    override fun toWeekViewEvent(): WeekViewEvent<Event> {
        // Build the styling of the event, for instance background color and strike-through
        val style = WeekViewEvent.Style.Builder()
            .setBackgroundColor(color)
            .setTextStrikeThrough(isCanceled)
            .build()

        // Build the WeekViewEvent via the Builder
        return WeekViewEvent.Builder<Event>(this)
            .setId(id)
            .setTitle(title)
            .setStartTime(startTime)
            .setEndTime(endTime)
            .setLocation(location)
            .setAllDay(isAllDay)
            .setStyle(style)
            .build()
    }

    // TODO: Need to implement this
//    override fun toString(): String {
//        return super.toString()
//    }
}